.github/workflows/build.yaml

```diff
@@ -86,14 +87,29 @@ jobs:
           ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_KEY }}
           ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.GPG_PASSPHRASE }}
         run: |
-          gradle -b galasa-managers-parent/build.gradle check publish \
+          set -o pipefail
+          gradle -b galasa-managers-parent/build.gradle check publish --info \
           -Dorg.gradle.jvmargs=-Xmx4096M \
           -PsourceMaven=https://development.galasa.dev/${{ env.ARGO_APP_BRANCH }}/maven-repo/extensions \
           -PcentralMaven=https://repo.maven.apache.org/maven2/ \
           -PtargetMaven=${{github.workspace}}/repo \
           -PjacocoEnabled=${{ inputs.jacocoEnabled }} \
-          -PisMainOrRelease=${{ inputs.isMainOrRelease }}
-
+          -PisMainOrRelease=${{ inputs.isMainOrRelease }} 2>&1 | tee build.log
+        
+      - name: Upload Gradle Build Log
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: gradle-build-log
+          path: build.log
+          
+      - name: Upload Jacoco Report
+        if: always()

```

Comment:
After discussing with a colleague, we determined that we should only do these on failure of the workflow, as you wouldn't bother to check the log if the run is successful. Please can we update these to `if: failure()` and set a retention time of 7 days?